sebex/release/state.py

```diff
@@ -201,14 +204,29 @@ def _build_plan(self, db: AnalysisDatabase, graph: DependentsGraph):
 
         # Here we go
         for project, dependency, relation in self._dependency_relations(db, graph):
+            # Don't bump nor update dependencies of prerelease packages (alpha etc.)
+            if projects[dependency].from_version._prerelease is not None and dependency not in source_names:
+                continue
             # We need to handle each dependency kind (version req, git, path) separately
             if relation.version_spec.is_version:
                 req: VersionRequirement = relation.version_spec.value
                 # We have to release a new version of dependent if its relation
                 # points to soon-to-be-outdated version of the dependency.
-                if (req.match(project.from_version) or req.match(_previous_version(project.to_version))) and not req.match(project.to_version):
-                    dep_bump = bumps[project.project].derive(project.from_version)
-                    bumps[dependency] = max(bumps[dependency], dep_bump)
+                release_new_version = ((req.match(project.from_version) or req.match(_previous_version(project.to_version))) and not req.match(project.to_version))
+                dependent_is_obsolete = not req.match(project.from_version) and not req.match(project.to_version)
+                if dependent_is_obsolete and dependency not in obsolete_pkg_updates.keys():
+                    obsolete_pkg_updates[dependency] = False
+                # Obsolete packages that are not directly dependent on a bumped package should be ignored
+                ignore_dependent = bumps[project.project] == Bump.STAY_AS_IS
+                update_dependent = update_obsolete and dependent_is_obsolete and not ignore_dependent
+                if update_dependent:
+                    obsolete_pkg_updates[dependency] = True
+
+                if release_new_version or update_dependent:
+                    # if a dependency of the package changed it's MINOR or MAJOR version then bump dependent by MINOR
+                    req_bump = Bump.MINOR if update_dependent else Bump.STAY_AS_IS
+                    version_bump = Bump.between(project.from_version, project.to_version)
+                    bumps[dependency] = max(req_bump, version_bump)
```

Comment:
I'm not sure if this logic is correct. I don't use `Bump.derive()`. Also there is probably a nicer way to write this.